### PR TITLE
rke: init at 1.4.1

### LINKED ISSUE #143217: RKE
**Project description**
Rancher Kubernetes Engine (RKE), an extremely simple, lightning fast Kubernetes distribution that runs entirely within containers. 

**Metadata**

* homepage URL: https://github.com/rancher/rke
* source URL: https://github.com/rancher/rke
* license: Apache-2.0 License 
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #143217

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/rke/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ buildGoModule, fetchFromGitHub, lib }:
```
Comment (figsoda) [inline pkgs/applications/networking/cluster/rke/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/applications/networking/cluster/rke/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "rancher";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-0tHTndcHemmh7XS3Wg943s0MsAzFHfnyBEMO3h/79K0=";
+  };
+
+  vendorHash = "sha256-zV1lrJhzrUAcEk6jYLCFrHcYw3CZart46qXErCTjZyQ=";
+
```
Comment (figsoda) [inline pkgs/applications/networking/cluster/rke/default.nix:15]:
```suggestion

  subPackages = [ "." ];

  ldflags = [ "-s" "-w" "-X=main.VERSION=v${version}" ];

```

File: pkgs/applications/networking/cluster/rke/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/rancher/rke";
+    description = "An extremely simple, lightning fast Kubernetes distribution that runs entirely within containers.";
```
Comment (figsoda) [inline pkgs/applications/networking/cluster/rke/default.nix:18]:
```suggestion
    description = "An extremely simple, lightning fast Kubernetes distribution that runs entirely within containers";
```

File: pkgs/applications/networking/cluster/rke/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    changelog = "https://github.com/rancher/rke/releases/tag/v${version}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ urandom ];
```
Comment (figsoda) [inline pkgs/applications/networking/cluster/rke/default.nix:21]:
```suggestion
    license = licenses.asl20;
    maintainers = with maintainers; [ urandom ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]
also 1.4.1 is out

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 203712` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>rke</li>
</ul>
</details>

File: pkgs/applications/networking/cluster/rke/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "rancher";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-H5T7Hr6/YL93lbCLOxfQ+Kmv3C+FS074418mHhBV3Bs=";
+  };
+
+  vendorHash = "sha256-zV1lrJhzrUAcEk6jYLCFrHcYw3CZart46qXErCTjZyQ=";
+
```
Comment (figsoda) [inline pkgs/applications/networking/cluster/rke/default.nix:15]: this will exclude the `codegen` binary
```suggestion

  subPackages = [ "." ];

```

**Review by figsoda — APPROVED** [review]

**Comment by zarelit** [comment]
yay! Thanks @urandom2 , @figsoda , @gador

**Pull request merged**